bus-mapping/src/evm/opcodes/number.rs

```diff
@@ -19,47 +19,60 @@ mod number_tests {
         };
 
         // Get the execution steps from the external tracer
-        let block =
-            TestContext::new_from_geth_data(new_single_tx_trace_code_at_start(&code).unwrap());
+        let block: GethData = TestContext::<2, 1>::new(
+            None,
+            account_0_code_account_1_no_code(code),
+            tx_from_1_to_0,
+            |block, _tx| block.number(0xcafeu64),
+        )
+        .unwrap()
+        .into();
 
-        let mut builder = block.new_circuit_input_builder();
-        builder.handle_tx(&block.eth_tx, &block.geth_trace).unwrap();
-
-        let mut test_builder = block.new_circuit_input_builder();
-        let mut tx = test_builder
-            .new_tx(&block.eth_tx, !block.geth_trace.failed)
+        let mut builder = BlockData::new_from_geth_data(block.clone()).new_circuit_input_builder();
+        builder
+            .handle_block(&block.eth_block, &block.geth_traces)
             .unwrap();
-        let mut tx_ctx = TransactionContext::new(&block.eth_tx, &block.geth_trace).unwrap();
+
+        let mut tx = builder.block.txs()[0].clone();
+
+        let mut tx_ctx = TransactionContext::new(
+            &block.eth_block.transactions[0],
+            &block.geth_traces[0],
+            true,
+        )
+        .unwrap();
 
         // Generate step corresponding to NUMBER
         let mut step = ExecStep::new(
-            &block.geth_trace.struct_logs[0],
+            &block.geth_traces[0].struct_logs[0],
             0,
-            test_builder.block_ctx.rwc,
+            builder.block_ctx.rwc,
             0,
         );
-        let mut state_ref = test_builder.state_ref(&mut tx, &mut tx_ctx);
+        let mut state_ref = builder.state_ref(&mut tx, &mut tx_ctx);
 
         // Add the last Stack write
         let number = block.eth_block.number.unwrap().as_u64();
-        state_ref.push_stack_op(
-            &mut step,
-            RW::WRITE,
-            StackAddress::from(1024 - 1),
-            eth_types::U256::from(number),
-        );
+        state_ref
+            .push_stack_op(
+                &mut step,
+                RW::WRITE,
+                StackAddress::from(1024 - 1),
+                eth_types::U256::from(number),
+            )
+            .unwrap();
 
         tx.steps_mut().push(step);
-        test_builder.block.txs_mut().push(tx);
+        builder.block.txs_mut().push(tx);
 
         // Compare first step bus mapping instance
         assert_eq!(
             builder.block.txs()[0].steps()[0].bus_mapping_instance,
-            test_builder.block.txs()[0].steps()[0].bus_mapping_instance,
+            builder.block.txs()[0].steps()[0].bus_mapping_instance,
```

Comment:
This is cheating! You have the same thing in both sides of the equality! :see_no_evil:

Reply:
Take a look at how the other tests for `gen_associated_ops` are implemented currently.  They no longer use a `test_builder`.  Instead they construct a single builder and do assertions directly on the resulting container of the builder.
I suggest following the same approach here.

Reply:
I'm stupid...